activerecord/lib/active_record/associations.rb

```diff
@@ -300,10 +300,10 @@ def association_instance_set(name, association)
     #
     # === A word of warning
     #
-    # Don't create associations that have the same name as instance methods of
-    # ActiveRecord::Base. Since the association adds a method with that name to
-    # its model, it will override the inherited method and break things.
-    # For instance, +attributes+ and +connection+ would be bad choices for association names.
+    # Don't create associations that have the same name as [instance methods](http://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-c-new) of
```

Comment:
I dont think we should remove that.
I think this is already a good improvement.
As a final change. can you: 
- update the link, to remove `#method-c-new` from it.

Thanks